client/state/data-layer/wpcom/sites/stats/post/index.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	POST_STATS_REQUEST,
+	POST_STATS_REQUEST_START,
+	POST_STATS_REQUEST_SUCCESS,
+	POST_STATS_REQUEST_FAILURE
+} from 'state/action-types';
+import { receivePostStat } from 'state/stats/posts/actions';
+
+export const requestPostStat = ( { dispatch }, { stat, siteId, postId } ) => {
+	dispatch( {
+		type: POST_STATS_REQUEST_START,
+		stat,
+		postId,
+		siteId
+	} );
+
+	return wpcom.site( siteId ).statsPostViews( postId, {
+		fields: stat
+	} ).then( data => {
+		if ( stat in data ) {
+			dispatch( receivePostStat( stat, siteId, postId, data[ stat ] ) );
+		}
+		dispatch( {
+			type: POST_STATS_REQUEST_SUCCESS,
+			stat,
+			siteId,
+			postId
+		} );
+	} ).catch( error => {
+		dispatch( {
+			type: POST_STATS_REQUEST_FAILURE,
+			stat,
+			siteId,
+			postId,
+			error
+		} );
+	} );
+};
```

Comment:
This is a good place to think about using the `BATCH` endpoint. I don't know how the post stat requests go out, but they are simple `GET` requests aren't they? In such a case we could queue these up in increments of say, 50ms, then send them all together in one.

```js
const queue = ( store, action ) => {
	fetchQueue.push( action );
	if ( fetchQueue.olderThan( 50 ) ) {
		requests = fetchQueue.flush();
		wpcom
			.batch( requests.map( toRequestUrl ) )
			.then( responses => responses.forEach( response.dispatch( receivePostStat( response ) ) ) );
	}
}

export default {
	[ POST_STATS_REQUEST ]: [ queue ],
}
```

please don't use this code. it's only meant to illustrate the idea